localstack/utils/aws/message_forwarding.py

```diff
@@ -47,7 +48,16 @@ def send_event_to_target(
             role_arn=role, service_principal=source_service, region_name=region
         )
     else:
-        clients = connect_to(region_name=region)
+        try:
+            from localstack_ext.config import ENFORCE_IAM
+
+            clients = (
+                connect_to(region_name=region)
+                if ENFORCE_IAM
+                else connect_to(aws_access_key_id=account_id, region_name=region)
+            )
+        except ImportError:
+            clients = connect_to(aws_access_key_id=account_id, region_name=region)
```

Comment:
I am not a fan of this at all. We should never make client creation dependent on the enforce IAM flag set - it should always be configured identically, with error handling if it fails due to a missing IAM permission. We should also not import pro code from community.

We have to ask us why this is necessary here. For all services which allow resource based policies (like lambda, sqs) it should not be necessary - they should parse the arn from the target themselves.

For those where it is necessary, it should always be conceptually the source account, not the target one (if cross account is possible).
We can allow this in the IAM engine itself - I can open a PR for it. Currently, it will recognize a call as being made from a service principal only if the access key id is `__internal__` - if this is too limiting, we can accept numeric account ids there as well.